### ISSUE TITLE
when redirecting host-pair gets out of sync

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1395]
```
	/* save old address for REDIRECTED_FROM notify */
	c->redirect.old_gw_address = c->remote->host.addr;
	/* update host_addr of other end, port stays the same */
	c->remote->host.addr = redirect_ip;
```

**Comment by cagney** [comment]
```
| "westnet-eastnet-ipv4-psk-ikev2": 192.1.2.45->192.1.2.23 oriented=no
| "westnet-eastnet-ipv4-psk-ikev2": 192.1.2.45->192.1.2.23 oriented=yes
| looking for host pair matching 192.1.2.23->192.1.2.45
| "westnet-eastnet-ipv4-psk-ikev2": 192.1.2.45->192.1.3.33 oriented=yes
EXPECTATION FAILED: entry connection connection_db_entries.host_pair@0x7f3e56455aa8 "westnet-eastnet-ipv4-psk-ikev2" is in the wrong bucket (check_hash_table() +145 programs/pluto/hash_table.c)
```

**Comment by cagney** [comment]
fixed by 60c4ce15bee4b97fff615f7b3ca78065b964a35d
